signer/src/main.rs

```diff
@@ -41,22 +48,30 @@ struct SignerArgs {
     /// pending migrations to the database on startup.
     #[clap(long)]
     migrate_db: bool,
+
+    #[clap(short = 'o', long = "output-format", default_value = "pretty")]
+    output_format: Option<LogOutputFormat>,
 }
 
 #[tokio::main]
 #[tracing::instrument(name = "signer")]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
-    // Initialize logging
-    // TODO(497): The whole logging thing should be revisited. We should support
-    //   enabling different layers, i.e. for pretty console, for opentelem, etc.
-    //sbtc::logging::setup_logging("info,signer=debug", false);
-    let _ = tracing_subscriber::fmt()
-        .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
-        .try_init();
-
     // Parse the command line arguments.
     let args = SignerArgs::parse();
 
+    // Initialize logging
+    let tracing_filter = tracing_subscriber::EnvFilter::from_default_env();
+    if let Some(LogOutputFormat::Json) = args.output_format {
+        tracing_subscriber::fmt()
+            .json()
+            .with_env_filter(tracing_filter)
+            .init();
```

Comment:
There is also `signer::logging::setup_logging(_, false)`, which is what I typically use for JSON logging

Reply:
[8f788e8](https://github.com/stacks-network/sbtc/pull/920/commits/8f788e8307b5a98edd2082386bfd8d0123bcf40a)

I liked that json better, but the pretty was _too_ pretty :laughing: